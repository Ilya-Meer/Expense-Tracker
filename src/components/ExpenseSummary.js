import React, { Component } from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import expenseTotal from '../selectors/expenseTotal';


class ExpenseSummary extends Component {
  constructor(props) {
    super(props)

    this.sumExpenses = this.sumExpenses.bind(this);
    this.formatTotal = this.formatTotal.bind(this);
  }

  sumExpenses(arr) {
    return numeral(expenseTotal(arr) / 100).format('$0,0.00');
  }

  formatTotal() {
    if (this.props.expenses.length === 0) {
      return `Viewing 0 expenses totalling ${this.sumExpenses(this.props.expenses)}`
    } else if (this.props.expenses.length === 1) {
      return `Viewing 1 expense totalling ${this.sumExpenses(this.props.expenses)}`
    } else {
      return `Viewing ${this.props.expenses.length} expenses totalling ${this.sumExpenses(this.props.expenses)}`
    }
  }

  render() {
    return (
      <div>
        <h2>{this.formatTotal()}</h2>
      </div>
    )
  }
}

export { ExpenseSummary };

const mapStateToProps = (state) => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseSummary);