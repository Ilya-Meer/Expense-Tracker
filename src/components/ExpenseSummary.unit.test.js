import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { ExpenseSummary } from './ExpenseSummary';

describe('Expense Summary Component', () => {
  it('should render properly with no expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
  })
  it('should render properly with one expense', () => {
    const expenses = [
      {
        id: '2',
        desc: 'Rent', 
        note: 'The rent was absurd', 
        amount: '10', 
        createdAt: moment(0).add('4', 'weeks').valueOf()
      }
    ]
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
  })
  it('should render properly with multiple expenses', () => {
    const expenses = [
      {
        id: '2',
        desc: 'Rent', 
        note: 'The rent was absurd', 
        amount: '10', 
        createdAt: moment(0).add('4', 'weeks').valueOf()
      },
      {
        id: '3',
        desc: 'Coffee', 
        note: 'I\'m just speechless...', 
        amount: '10', 
        createdAt: moment(0).subtract('4', 'days').valueOf()
      }
    ]
    const wrapper = shallow(<ExpenseSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
  })
})