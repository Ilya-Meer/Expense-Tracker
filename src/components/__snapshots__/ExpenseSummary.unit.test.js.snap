// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expense Summary Component should render properly with multiple expenses 1`] = `
<div>
  <h2>
    Viewing 2 expenses totalling $0.20
  </h2>
</div>
`;

exports[`Expense Summary Component should render properly with no expenses 1`] = `
<div>
  <h2>
    Viewing 0 expenses totalling $0.00
  </h2>
</div>
`;

exports[`Expense Summary Component should render properly with one expense 1`] = `
<div>
  <h2>
    Viewing 1 expense totalling $0.10
  </h2>
</div>
`;
